emukit/core/loop/loop_state.py

```diff
@@ -33,32 +34,44 @@ def update(self, results: List[UserFunctionResult]) -> None:
     @property
     def X(self) -> np.ndarray:
         """
-        :return: Function inputs for all function evaluations
-                 in a 2d array: number of points by input dimensions.
+        :return: Function inputs for all function evaluations in a 2d array: number of points by input dimensions.
         """
         return np.array([result.X for result in self.results])
 
     @property
     def Y(self) -> np.ndarray:
         """
-        :return: Function outputs for all function evaluations
-                 in a 2d array: number of points by output dimensions.
+        :return: Function outputs for all function evaluations in a 2d array: number of points by output dimensions.
         """
         return np.array([result.Y for result in self.results])
 
+    @property
+    def cost(self) -> np.ndarray:
+        """
+        :return: Cost for evaluating the function in a 2d array: number of points by cost dimensions.
+        """
+        return np.array([result.cost for result in self.results])
```

Comment:
Is it guaranteed that every result object has a cost field, and that this field contains something sensible? Even for non-cost-aware loops?

Reply:
It defaults to None so you would get an array of None's here